src/reactive/extMethods/asRouterObservable.js

```diff
@@ -0,0 +1,30 @@
+// notice_start
+/*
+ * Copyright 2015 Dev Shop Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// notice_end
+
+import Observable from '../Observable';
+import { Guard } from '../../system';
+import RouterObservable from '../RouterObservable';
+
+Observable.prototype.asRouterObservable = function(router) {
+    Guard.isDefined(router, "router must be defined");
+    var source = this;
+    var subscribe =  observer => {
```

Comment:
could be more terse and just have `observer => source.subscribe(observer);`  as you have above in another file